plugins/inputs/statsd/statsd.go

```diff
@@ -763,6 +764,10 @@ func (s *Statsd) parseName(bucket string) (name string, field string, tags map[s
 	}
 
 	name = bucketparts[0]
+	name = strings.ReplaceAll(name, " ", "_")
+	name = strings.ReplaceAll(name, "/", "-")
+	allowedChars := regexp.MustCompile(`[^a-zA-Z_\-0-9\.;=]`)
+	name = allowedChars.ReplaceAllString(name, "")
```

Comment:
Could you add a unit test covering the sanitizing?

My first reaction is to ask for a setting that enables the sanitizing, but it looks like telegraf is dropping metric data with names that need to be sanitized so it doesn't make sense not to sanitize. Is that right?

Reply:
Will do the unit tests.

And correct, we should be sanitizing so users actually get the data to avoid dropping or garbling the data.